rust_snuba/src/consumer.rs

```diff
@@ -286,10 +292,21 @@ pub fn consumer_impl(
 
     let mut handle = processor.get_handle();
 
-    ctrlc::set_handler(move || {
-        handle.signal_shutdown();
-    })
-    .expect("Error setting Ctrl-C handler");
+    match rebalance_delay_secs {
+        Some(secs) => {
+            ctrlc::set_handler(move || {
+                rebalancing::delay_kafka_rebalance(secs);
+                handle.signal_shutdown();
```

Comment:
Am I correct in assuming that this closure that is passed in runs in a separate thread, so the main loop is not blocked while it is sleeping before signaling shutdown?

Reply:
From the documentation it seems to be the case

https://docs.rs/ctrlc/latest/ctrlc/fn.set_handler.html

```
Register signal handler for Ctrl-C.

Starts a new dedicated signal handling thread. Should only be called once, typically at the start of your program.
```

Reply:
Nice!